src/guc.c

```diff
@@ -1067,6 +1081,19 @@ _guc_init(void)
 							 /* assign_hook= */ NULL,
 							 /* show_hook= */ NULL);
 
+	DefineCustomIntVariable(/* name= */ MAKE_EXTOPTION("debug_bgw_scheduler_exit_status"),
```

Comment:
Nit: you don't need to expose full GUCs to achieve this debug functionality.  You can just check for a settings variable. Then the code can also be localized to where it is needed.